src/tests/ftest/cart/test_group_np_cli.c

```diff
@@ -46,30 +91,56 @@ test_run(void)
 	crt_rpc_t		*rpc_req = NULL;
 	int			 i;
 	int			 rc = 0;
+	uint32_t		*_cg_ranks;
+	int			 _cg_num_ranks;
 
-	if (test_g.t_save_cfg) {
-		rc = crt_group_config_path_set(test_g.t_cfg_path);
-		D_ASSERTF(rc == 0, "crt_group_config_path_set failed %d\n", rc);
-	}
+	if (test_g.t_skip_init) {
+		DBG_PRINT("Skipping init stage.\n");
 
-	tc_cli_start_basic(test_g.t_local_group_name,
-			   test_g.t_remote_group_name,
-			   &grp, &rank_list, &test_g.t_crt_ctx[0],
-			   &test_g.t_tid[0], test_g.t_srv_ctx_num,
-			   test_g.t_use_cfg, NULL);
+	} else if (!test_g.t_skip_init) {
 
-	rc = sem_init(&test_g.t_token_to_proceed, 0, 0);
-	D_ASSERTF(rc == 0, "sem_init() failed.\n");
+		if (test_g.t_save_cfg) {
+			rc = crt_group_config_path_set(test_g.t_cfg_path);
+			D_ASSERTF(rc == 0,
+				  "crt_group_config_path_set failed %d\n", rc);
+		}
 
-	/* register RPCs */
-	rc = crt_proto_register(&my_proto_fmt_test_group2);
-	D_ASSERTF(rc == 0, "crt_proto_register() failed. rc: %d\n",
-			rc);
+		tc_cli_start_basic(test_g.t_local_group_name,
+				   test_g.t_remote_group_name,
+				   &grp, &rank_list, &test_g.t_crt_ctx[0],
+				   &test_g.t_tid[0], test_g.t_srv_ctx_num,
+				   test_g.t_use_cfg, NULL);
+
+		rc = sem_init(&test_g.t_token_to_proceed, 0, 0);
+		D_ASSERTF(rc == 0, "sem_init() failed.\n");
+
+		/* register RPCs */
+		rc = crt_proto_register(&my_proto_fmt_test_group2);
+		D_ASSERTF(rc == 0, "crt_proto_register() failed. rc: %d\n",
+			  rc);
+
+		/* Process the --rank option, e.g., --rank 1,2-4 */
+		if (test_g.cg_num_ranks > -1) {
```

Comment:
why > -1 instead of > 0?

Reply:
Is it possible to have a rank list of length 0?  I wanted to distinguish between a 0 rank list length and a NULL/undefined length.

Reply:
17af1bf3c715de43e313528cc240abd69a8346f2